js/ui/searchbar.js

```diff
@@ -145,19 +160,25 @@ define([
       var options = this.currentSearch;
 
       chrome.fileSystem.getDisplayPath(nodeEntry, function(path) {
-        var file = new File(nodeEntry);
         if (!options.running) {
           return c();
         }
 
-        var printResult = function(index, str) {
-          self.appendToResults(self.formatResultCode(index, str));
-        };
+        var file = new File(nodeEntry);
 
         file.read(function(err, data) {
+          if (!options.running) {
```

Comment:
this was also needed in order to make sure short circuiting cuts off at the right number.